src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/JsonPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,242 @@
+
+            layout.setVerticalGroup(layout.createSequentialGroup()
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(numberOfPayloadsLabel)
+                                    .addComponent(numberOfPayloadsSpinner))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(jsonFieldLabel)
+                                    .addComponent(jsonFieldScrollPane)));
+        }
+
+        @Override
+        protected JsonPayloadGenerator getPayloadGenerator() {
+            if (!validate()) {
+                return null;
+            }
+            return new JsonPayloadGenerator(jsonTextArea.getText(), numberOfPayloadsSpinner.getValue());
+        }
+
+        @Override
+        public void init(MessageLocation messageLocation) {
+            getJsonTextArea().setText(messageLocation.getValue());
+        }
+
+        @Override
+        public JPanel getComponent() {
+            return fieldsPanel;
+        }
+
+        @Override
+        public void setPayloadGeneratorUI(JsonPayloadGeneratorUI payloadGeneratorUI) {
+            oldGenerator = payloadGeneratorUI;
+            numberOfPayloadsSpinner.setValue(payloadGeneratorUI.getPayloadGenerator().getNumberOfPayloads());
+            jsonTextArea.setText(payloadGeneratorUI.getPayloadGenerator().getJson());
+            jsonTextArea.discardAllEdits();
+        }
+
+        @Override
+        public JsonPayloadGeneratorUI getPayloadGeneratorUI() {
+            if (oldGenerator != null) {
+                return oldGenerator;
+            }
+
+            return new JsonPayloadGeneratorUI(getPayloadGenerator());
+        }
+
+        @Override
+        public void clear() {
+            oldGenerator = null;
+
+            getJsonTextArea().setText("");
+            jsonTextArea.discardAllEdits();
+            numberOfPayloadsSpinner.setValue(1);
+        }
+
+        @Override
+        public boolean validate() {
+            if (oldGenerator != null &&
+                    getJsonTextArea().getText().equals(oldGenerator.getPayloadGenerator().getJson()) &&
+                    numberOfPayloadsSpinner.getValue() == oldGenerator.getPayloadGenerator().getNumberOfPayloads()) {
+                return true;
+            }
+
+            if (getJsonTextArea().getDocument().getLength() == 0) {
+                return false;
+            }
+            if (isInvalidJson(getJsonTextArea().getText())) {
+                return false;
+            }
+
+            oldGenerator = null;
+            return true;
+        }
+
+        private boolean isInvalidJson(String json) {
+            try {
+                new JsonParser().parse(json);
+                return false;
+            } catch (Exception e) {
+                return false;
```

Comment:
`true`?

Reply:
I will write a test. Thanks for catching this one :-/

Reply:
No problem.